src/python/dart/web/api/engine.py

```diff
@@ -86,7 +86,12 @@ def action_checkout(action):
     if len(results) == 3:
         return results
 
-    action = workflow_service().action_checkout(action)
+    action = None
+    try:
+        action = workflow_service().action_checkout(action)
+    except Exception as err:
+        return {'result': str(err)}, 529
```

Comment:
Should this throw a `DartRequestException`? https://github.com/RetailMeNotSandbox/dart/blob/master/src/python/dart/model/exception.py#L1